smart_campus/app/models.py

```diff
@@ -1,3 +1,222 @@
 from django.contrib.gis.db import models
+from django.contrib.auth.base_user import (
+    BaseUserManager, AbstractBaseUser
+)
 
-# Create your models here.
+class Beacon(models.Model):
+    beacon_id = models.CharField(max_length=200, primary_key=True)
+    name = models.CharField(max_length=200)
+    description = models.CharField(max_length=200, blank=True)
+    location = models.GeometryField(srid=4326, null=True)
+    owner_group = models.ForeignKey('UserGroup', null=True, on_delete=models.SET_NULL)
+    station = models.ForeignKey('Station', null=True, on_delete=models.SET_NULL)
+
+    def __str__(self):
+        return self.name
+
+class UserManager(BaseUserManager):
+    def create_user(self, email, password=None):
+        """
+        Creates and saves a User with the given email and password
+        """
+        if not email:
+            raise ValueError('Users must have an email address')
+
+        user = self.model(
+            email=self.normalize_email(email),
+        )
+
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+class User(AbstractBaseUser):
+    email = models.EmailField(max_length=255, unique=True, primary_key=True)
+    password = models.CharField(max_length=128)
+    nickname = models.CharField(max_length=254, blank=True)
+    role = models.ForeignKey('Role', null=True, on_delete=models.SET_NULL)
+    group = models.ForeignKey('UserGroup', null=True, on_delete=models.SET_NULL)
+
+    experience_point = models.IntegerField(default=0)
+    earned_coins = models.IntegerField(default=0)
+
+    answered_questions = models.ManyToManyField('Question')
+    visited_beacons = models.ManyToManyField('Beacon')
+    favorite_stations = models.ManyToManyField('Station')
+
+    received_rewards = models.ManyToManyField(
+        'Reward',
+        through='UserReward',
+    )
+
+    # custom model manager for model 'User'
+    objects = UserManager()
+
+    USERNAME_FIELD = 'email'
+
+    def can(self, permissions):
+        return self.role is not None and \
+            (self.role.permissions & permissions) == permissions
```

Comment:
```python
return (self.role is not None and
            (self.role.permissions & permissions) == permissions)
```

According to PEP8, parentheses should be used for breaking lines.

The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.